samples/bluetooth/peripheral_fast_pair/src/main.c

```diff
@@ -43,11 +44,17 @@ static void advertising_start(void)
 {
 	int err = bt_adv_helper_adv_start(fp_adv_mode);
 
+	int ret = k_work_cancel_delayable(&fp_discoverable_adv_timeout);
```

Comment:
You have two variables for holding return values/error codes. Is it intended?

I recommend that you split the variable declaration and its assignment and have the ASSERT checks in one code block (without line separator at L48)

Reply:
It's intended here. I wanted to separate asserts related to work states from handling error values returned by other functions (the returned value often describes state of the work, so I would prefer calling it ret, not err too)

Reply:
> I recommend that you split the variable declaration and its assignment and have the ASSERT checks in one code block (without line separator at L48)

In the sample code there are already many cases where we combine variable declaration and assignment. Could we keep it as it is?

Reply:
If they don't break the return value formatting that I described in this https://github.com/nrfconnect/sdk-nrf/pull/8596#discussion_r960583145 comment, then I wouldn't change it in the scope of this PR.

Please consider separating the variable declarations from assignments at some point in future PRs. I believe it improves readability of the code

Reply:
> Please consider separating the variable declarations from assignments at some point in future PRs. I believe it improves readability of the code

I would say that separating the variable declarations from assignments is not the only good practice. Declaring variables right before using it also has some advantages. Both approaches have pros and cons. I think that if we keep code of a given sample/subsystem more or less consistent, we should be fine (our coding standard seems actually not to impose this).

> If they don't break the return value formatting that I described in this [#8596 (comment)](https://github.com/nrfconnect/sdk-nrf/pull/8596#discussion_r960583145) comment, then I wouldn't change it in the scope of this PR.

The empty line after declaration is imposed by compliance, so we cannot simply omit it. Separate declaration could help here, but as I mentioned before, this approach also has some cons - e.g. may introduce risk of using variable before it's initialized.